src/main/java/carpet/script/language/Loops.java

```diff
@@ -140,7 +143,45 @@ public static void apply(Expression expression)
             c.setVariable("_i", _iter);
             return (cc, tt) ->  ret;
         });
+//=========
+        expression.addFunction("zip", (lv) ->
+        {
+            if (lv.stream().allMatch(v->(v instanceof AbstractListValue))){
+                return new LazyListValue(){
+                    List<AbstractListValue> b=lv.stream().map(v->(AbstractListValue)v).collect(Collectors.toList());
+                    List<Iterator<Value>> a=lv.stream().map(v->((AbstractListValue)v).iterator()).collect(Collectors.toList());
+                    
+
+                    @Override
+                    public boolean hasNext() {
+                        return a.stream().allMatch(Iterator::hasNext);
+                    }
+
+                    @Override
+                    public Value next() {
+                        return ListValue.wrap(a.stream().map(Iterator::next));
```

Comment:
worrying if some of the element of this list is the same iterator.
Then these iterators run the risk of overstepping the bounds.
But it is a classic use of zip to put the same object in the argument multiple times.
If it does go out of bounds I can't squeeze the elements I've taken out back into the iterator 🤔

Reply:
valid concern, because peeps might want to use it like a = range(10); zip(a, a); btw - that works in python

Reply:
i changed it to zip_longest and fill the void with Null...😅